source/common/http/http1/codec_impl.cc

```diff
@@ -64,6 +66,16 @@ const std::string StreamEncoderImpl::CRLF = "\r\n";
 // Last chunk as defined here https://tools.ietf.org/html/rfc7230#section-4.1
 const std::string StreamEncoderImpl::LAST_CHUNK = "0\r\n";
 
+CodecStats::CodecStats(const CodecStatNames& stat_names, Stats::Scope& scope)
```

Comment:
This still does a lot of work when each new connection is made. I think we should be creating a stat struct (potentially using the macros) that is owned by the HCM config and/or cluster? This would be constructed once with the right prefix and then used over and over? Any reason to not go this approach?